### ISSUE TITLE
PWM TTL For laser

### ISSUE THREAD

**Issue by Ainmimdal** [makerbase-mks/MKS-Robin-Nano-V3.X#8]
Hi, I need to supply 5v pwm signal to control my laser. Supposedly this board is said to have PWM TTL support compared to the robin nano V2, but I checked my board and the only empty pins I see are the three next to the fan LEDs. The silkscreen on the board says PA0 PE5 and PB0 which is the same for the heaters. I don't see any spindle/laser definitions in the example configs either.

**Comment by Ainmimdal** [comment]
I tried using the PB0 and the PWM works fine for now. But can anyone confirm the voltage on this pin?

**Comment by Ainmimdal** [comment]
After reading issue #7, Using TX3 (PB10) pin works. I still don't understand why the advertised PWM pins are the same pin for heaters.

**Comment by MKS-Sean** [comment]
![image](https://user-images.githubusercontent.com/56996910/109384234-5ebb8d80-7926-11eb-9947-0703d994ec7d.png)
Hello, what I marked with a red box is the TTL pin that drives the MOS tube, which can be configured as a laser control pin

**Comment by Ainmimdal** [comment]
Yeah, I tried those pins but the laser only goes up to 60% with those pins. Any components connected to the heater port will turn on if I use those pins. My solution is using the UART pins which outputs pwm correctly.

**Comment by moibatista** [comment]
I have the MKS Robin Nano v3, with original configuration. I don't change anything.
How I can use de TTL pin? How I can active this pins? I need to know the Gcode to control this pin.
With M3 say me incorrect command.

**Comment by Ainmimdal** [comment]
I use TX3 pin (next to the esp) and in marlin, you have to find the file pins_MKS_ROBIN_NANO_V3.h, find #define SPINDLE_LASER_PWM_PIN and change to #define SPINDLE_LASER_PWM_PIN PB10 . You can search more on the internet about changing the laser settings in configuration_adv.h

**Comment by CLP58** [comment]
> I use TX3 pin (next to the esp) and in marlin, you have to find the file pins_MKS_ROBIN_NANO_V3.h, find #define SPINDLE_LASER_PWM_PIN and change to #define SPINDLE_LASER_PWM_PIN PB10 . You can search more on the internet about changing the laser settings in configuration_adv.h

I tried using the TTL PWM pin as they say, but found the same thing as you, only 60% power. How did you fix it please? I used the UART the same way you did and only had 60% power.

**Comment by danorder** [comment]
did you change the jumper to 5v.
